### PR TITLE
Attachment to new tab

### PR DESCRIPTION
**Pull request by MaloMn**
I think the behavior when viewing attachment should change: when clicking on an image in a bugnote, it displays it in the same tab. I think it would be better to redirect this full display of the image to a new tab. That would allow you to go back and forth between seeing the image and seeing the bugnote text associated.

This new redirection would be effective for both clicking on the icon or on the name of the attachment (as shown below), and also when clicking on an image preview in a bugnote, or a preview of an audio / video.

image

### CODE REVIEW

**Comment by atrol** [comment]
Worth to read in this context https://mantisbt.org/forums/viewtopic.php?f=2&t=25561

**Comment by MaloMn** [comment]
Thanks for the forum link.

From what I understand:
* It's something that should be an option, and not something permanent,
* It's unclear whether it should behave the same for pdf / audio / images / text files.

I don't think setting an option for each file type is a good idea (I agree with your argument that it would confuse users), and I also think that having an option for all file types is a good thing. I don't think it woud be right to open pdf files in a new tab, and not images. In both cases, you have to use the browser "Go back" button if it does not open in a new tab.
I also don't think that the question about whether the browser can handle displaying the attachment is valid, because whether it is in a new tab or not, the browser will behave the same. The only major change in my view is if it can't display the attachment : it will open a new tab and close it immediately, before opening the dialog box to open the attachent in another software.

I'll reply in this forum with a link to this Pull Request.

Following up from what I've said, I added a constant in config_default_inc.php, and a function in print_api.php that returns the `target` tag, filleed with either *_self* or *_blank*. I'm not sure whether it's the right way to do it, but it works on my local Mantis installation.

File: core/print_api.php [hunk 1]
```diff
@@ -2125,6 +2125,17 @@ function get_filesize_info( $p_size, $p_unit ) {
 	return sprintf( lang_get( 'max_file_size_info' ), number_format( $p_size ), $p_unit );
 }
 
+/**
+ * Returns target tag to be added in attachment links
```
Comment (dregad) [inline core/print_api.php:2129]: *target* is not a tag, it's an attribute

File: core/print_api.php [hunk 2]
```diff
@@ -2125,6 +2125,17 @@ function get_filesize_info( $p_size, $p_unit ) {
...
+ * @return string
+ */
+function get_attachment_link_target() {
```
Comment (dregad) [inline core/print_api.php:2132]: By convention, we usually prefix functions with the API's name, so this should be called *print_attachment_link_target()*

File: config_defaults_inc.php [hunk 3]
```diff
@@ -2066,6 +2066,13 @@
  */
 $g_wrap_in_preformatted_text = ON;
 
+/**
+ * Controls whether attachment links redirect users t a new tab or not
+ * If turned off, a click on a link will not redirect to a new tab
+ * @global integer $g_attachment_to_new_tab
+ */
+$g_attachment_to_new_tab = OFF;
+
```
Comment (dregad) [inline config_defaults_inc.php:2075]: I believe it would make more sense to move this to the *Bug Attachments Settings* section

**Review by dregad — CHANGES_REQUESTED** [review]

**Comment by dregad** [comment]
@MaloMn as you have not responded to the review, I'm going to close this PR without merge. Feel free to reopen after making the required changes.

**Comment by dregad** [comment]
Replaced by #1994

**Pull request closed without merging**